src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/RazorCodeBlockFoldingProvider.cs

```diff
@@ -32,12 +33,18 @@ public async Task<ImmutableArray<FoldingRange>> GetFoldingRangesAsync(DocumentCo
 
                 // Directives remove the "@" but for collapsing we want to keep it for users.
                 // Shows "@code" instead of "code".
-                CollapsedText = "@" + codeBlock.DirectiveDescriptor.Directive 
+                CollapsedText = "@" + codeBlock.DirectiveDescriptor.Directive
             };
 
             builder.Add(foldingRange);
         }
 
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+
+        var razorFileSyntaxWalker = new RazorFileFoldingRangeSyntaxWalker(codeDocument.Source);
```

Comment:
Based on the conversation in our meeting just now, I guess this should move to its own new folding range provider. I didn't even notice it was in the code block folding range provider :)